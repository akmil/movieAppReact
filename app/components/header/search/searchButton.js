import React from 'react';
import { Button } from '../../common/button';
// import { withRouter } from 'react-router'

export const SearchButton = ({ onClick }) => <div className={'search-button'}>
        <Button onClick={onClick}>search
        </Button>
    </div>;
