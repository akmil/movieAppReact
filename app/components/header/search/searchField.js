import React from 'react';

export const SearchField = ({val}) => {
    return (
        <div className={'search-field-box'}>
            <label className={'search-field-label'}>Find your movie</label>
            <input placeholder={`Hey, ${val}`} className={'search-field-input'}/>
        </div>
    );
};

