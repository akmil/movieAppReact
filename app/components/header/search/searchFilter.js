import React from 'react';

export const SearchFilter = ({name}) => {
    return (
        <div className='search-filter-box'>
            Search by
            <ul className={'search-filter-list'}>
                <li>{`Title ${name}`}</li>
                <li>{`Genre ${name}`}</li>
            </ul>
        </div>
    );
};
