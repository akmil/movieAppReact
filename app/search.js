import React from 'react';
import Header from './components/header/header';
import ResultsBox from "./components/resultsBox/resultsBox";

class Search extends React.Component {

    componentDidMount(){
        // console.log('Search match.params', this.props.match.params);
    }

    render() {
        return (
            <div className={`main-layout-wrapper`}>
                <Header query={this.props.match.params.query}></Header>
                <ResultsBox></ResultsBox>
            </div>
        )
    }
}

export default Search;