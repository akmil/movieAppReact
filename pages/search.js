import React from 'react';
import Header from '../app/components/header/header';
import ResultsBox from "../app/components/resultsBox/resultsBox";

class Search extends React.Component {
    static getInitialProps = (ctx) => {
        const { query } = ctx;
        const pathToSearch = `${query.search}&searchBy=${query.searchBy}`;
        const userAgent = ctx.req ? ctx.req.headers['user-agent'] : 'someBrowser';

        console.log('ctx', ctx.asPath);
        console.log('query', query);
        // console.log('ctx-headers', ctx.req.headers['user-agent']);
        return {
            query,
            pathToSearch,
            userAgent
        }
    };

    componentDidMount(){
        console.log('Search query', this.props.query);
        console.log('Search userAgent', this.props.userAgent);
    }

    render() {
        return (
            <div className={`main-layout-wrapper`}>
                <Header query={this.props.pathToSearch} userAgent={this.props.userAgent}></Header>
                <ResultsBox></ResultsBox>
            </div>
        )
    }
}

export default Search;
