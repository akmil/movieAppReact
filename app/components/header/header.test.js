import React from 'react';
// import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from "./header";
import { shallow, mount } from 'enzyme'
import sinon from 'sinon'
import configureStore from 'redux-mock-store';
import {SearchField} from "./search/searchField";
const initialState = {runtime: {}, movies:[]};
const mockStore = configureStore();
const dispatch = jest.fn();

describe('Header', function () {
    it('should render without errors', () => {
        const wrapper = shallow(
            <Header dispatch={dispatch}
                    store={mockStore(initialState)}
            />
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('should render without errors', () => {
        const wrapper = shallow(
            <Header dispatch={dispatch}
                    store={mockStore(initialState)}
            />
        ).dive();

        wrapper.find(SearchField).simulate('change', {
            target: {
                value: 'hello world'
            }
        });

        expect(wrapper.state()).toEqual({"sortBy": "title", "text": "hello world"})
    });

    /*it('should render without errors', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <Header dispatch={dispatch}
                    store={store}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });*/

    // it('should be dispatch Header', function() {
    //     const wrapper = shallow(<Header dispatch={dispatch} store={mockStore({ runtime: {}, movies:[] })}/>);
    // });

    it('should render an `.logo`', () => {
        const wrapper = shallow(
            <Header dispatch={dispatch}
                    store={mockStore(initialState)}
            />
        ).dive();
        console.log('wrapper' , wrapper.find('.logo'))
        // expect(wrapper.find('.logo')).to.equal({length:1});
    });

   /* it('should render children when passed in', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.contains(<div className="header-wrapper" />)).to.equal(true);
    })*/

    /*it('simulates click events', () => {
        const onButtonClick = sinon.spy();
        const wrapper = shallow(<Foo onButtonClick={onButtonClick} />);
        wrapper.find('button').simulate('click');
        expect(onButtonClick.calledOnce).to.equal(true);
    });*/
});
