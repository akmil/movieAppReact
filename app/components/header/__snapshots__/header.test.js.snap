// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without errors 1`] = `
<header>
  <div
    className="header-wrapper"
  >
    <form
      action="/search/\${query}"
      className="form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="logo"
      >
        netfixroulette
      </div>
      <SearchField
        onChange={[Function]}
        val="Search by name"
      />
      <div
        className="search-filter"
      >
        <div
          className="search-filter-box"
        >
          Search by
          <ul
            className="search-filter-list"
          >
            <li
              className="active"
              onClick={[Function]}
            >
              Title
            </li>
            <li
              className=""
              onClick={[Function]}
            >
              Genre
            </li>
          </ul>
        </div>
        <SearchButton />
      </div>
    </form>
  </div>
</header>
`;

exports[`Header should render without errors 2`] = `
<Connect(Header)
  dispatch={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header
    dispatch={[MockFunction]}
    fetchMovie={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    totalMovies={0}
  >
    <header>
      <div
        className="header-wrapper"
      >
        <form
          action="/search/\${query}"
          className="form-horizontal"
          onSubmit={[Function]}
        >
          <div
            className="logo"
          >
            netfixroulette
          </div>
          <SearchField
            onChange={[Function]}
            val="Search by name"
          >
            <div
              className="search-field-box"
            >
              <label
                className="search-field-label"
              >
                Find your movie
              </label>
              <input
                className="search-field-input"
                onChange={[Function]}
                placeholder="Hey"
              />
            </div>
          </SearchField>
          <div
            className="search-filter"
          >
            <div
              className="search-filter-box"
            >
              Search by
              <ul
                className="search-filter-list"
              >
                <li
                  className="active"
                  onClick={[Function]}
                >
                  Title
                </li>
                <li
                  className=""
                  onClick={[Function]}
                >
                  Genre
                </li>
              </ul>
            </div>
            <SearchButton>
              <div
                className="search-button"
              >
                <Button>
                  <button
                    style={
                      Object {
                        "background": "#F44336",
                        "border": "none",
                        "color": "white",
                        "padding": "10px",
                      }
                    }
                  >
                    search
                  </button>
                </Button>
              </div>
            </SearchButton>
          </div>
        </form>
      </div>
    </header>
  </Header>
</Connect(Header)>
`;
